### PR TITLE
Implement configuration class

### LINKED ISSUE #80: Generate section related to configuration parameters of Record Manager
As a developer, I would like a section visible from README.md showing all the backend configuration parameters. Those parameters should be generated from Spring annotations related to configuration. There is an annotation processor plugin that can be used for this purpose.

 It should be done similarly to https://github.com/kbss-cvut/termit-docker?tab=readme-ov-file#further-termit-configuration. It is generated by an annotation processor (see example in https://github.com/kbss-cvut/termit/blob/22ffaed4a221679b2598aef0b11420243a27f123/pom.xml#L427), while generated values are taken from spring annotations 
`@Value` or `@ConfigurationProperties` (see example in https://github.com/kbss-cvut/termit/blob/master/src/main/java/cz/cvut/kbss/termit/util/Configuration.java)

A/C:
- configuration parameters are generated with build into .md file
- The .md file is included/referenced in README.md
- It is considered if it is useful to reference this .md file in record-manager-ui

### PR DESCRIPTION
**Pull request by palagdan**
Resolve #80

### CODE REVIEW

**Comment by palagdan** [comment]
@blcham

I created a configuration class that assigns all properties to variables. During the compile stage, a plugin generates a configuration.md file in the doc folder. Readme.md is updated too.

File: src/main/java/cz/cvut/kbss/study/util/Configuration.java [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+    /**
+     * Url of the website for password reset emails. e.g. https://study.com/ (must have "/" at the end)
+     */
+    String appContext;
+
+    /**
+     * Driver to manage triple stores
+     */
+    String driver;
```
Comment (blcham) [inline src/main/java/cz/cvut/kbss/study/util/Configuration.java:16]:
```suggestion
    String persistenceDriver;
```

Comment (blcham) [reply src/main/java/cz/cvut/kbss/study/util/Configuration.java:16]: better name for what it is.

File: src/main/java/cz/cvut/kbss/study/util/Configuration.java [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+        String from;
+
+        /**
+         * Email cc addresses where all invitations will be sent. For more use delimiter "," (can remain empty)
+         */
+        String replyTo;
+
+        /**
+         *
+         */
```
[the remark this answers is not in the document]
Comment (palagdan) [reply src/main/java/cz/cvut/kbss/study/util/Configuration.java]: done

File: src/main/java/cz/cvut/kbss/study/util/Configuration.java [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+        String cc;
+
+        /**
+         *
+         */
```
[the remark this answers is not in the document]
Comment (palagdan) [reply src/main/java/cz/cvut/kbss/study/util/Configuration.java]: done

File: src/main/java/cz/cvut/kbss/study/util/Configuration.java [hunk 4]
```diff
@@ -0,0 +1,433 @@
+package cz.cvut.kbss.study.util;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@ConfigurationProperties(prefix = "")
+public class Configuration {
+
+    /**
+     * Url of the website for password reset emails. e.g. https://study.com/ (must have "/" at the end)
```
[the remark this answers is not in the document]
Comment (palagdan) [reply src/main/java/cz/cvut/kbss/study/util/Configuration.java]: done

File: src/main/java/cz/cvut/kbss/study/util/Configuration.java (hunk 1 above)
Comment (palagdan) [reply src/main/java/cz/cvut/kbss/study/util/Configuration.java]: done

File: README.md [hunk 5]
```diff
@@ -15,13 +15,8 @@ The system can be split into two parts. __Main application__ provides management
...
-| ```RECORDS_ALLOWEDREJECTREASON```                  | Allow user to enter reason when rejecting records. Default value: `false.`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
+The table lists environment variables that can be passed to the Record Manager backend through `docker-compose.yml`, an [env_file](https://docs.docker.com/compose/compose-file/compose-file-v3/#env_file), 
+or the command line. For more details, see the [Configuration documentation](./doc/configuration.md).
```
[the remark this answers is not in the document]
Comment (palagdan) [reply README.md:19]: done

File: src/main/java/cz/cvut/kbss/study/util/Configuration.java (hunk 1 above)
Comment (blcham) [reply src/main/java/cz/cvut/kbss/study/util/Configuration.java:16]: can't we change it there as well?

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (blcham) [reply README.md:19]: i consider this comment done

**Comment by blcham** [comment]
@palagdan please incorporate following as well:

https://github.com/lukaskabc/spring-boot-configuration-docgen/issues/2#issuecomment-2506317121

File: Dockerfile [hunk 6]
```diff
@@ -8,6 +8,8 @@ RUN mvn -B de.qaware.maven:go-offline-maven-plugin:resolve-dependencies
 
 COPY src src
 
+RUN mkdir /record-manager/doc
```
Comment (blcham) [inline Dockerfile:11]: @palagdan please incorporate the following:

https://github.com/lukaskabc/spring-boot-configuration-docgen/issues/2#issuecomment-2506317121

**Review by blcham — CHANGES_REQUESTED** [review]
See my comments.

File: Dockerfile (hunk 6 above)
Comment (palagdan) [reply Dockerfile]: done

File: src/main/java/cz/cvut/kbss/study/util/Configuration.java (hunk 1 above)
Comment (palagdan) [reply src/main/java/cz/cvut/kbss/study/util/Configuration.java]: done

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (palagdan) [reply README.md:19]: I looked into some options for automatically including the content of one .md file in another, but Markdown does not support this feature. See: https://stackoverflow.com/questions/4779582/markdown-and-including-multiple-files.

**Pull request merged**